packages/js-moi-manifest/src.ts/manifest.ts

```diff
@@ -13,10 +14,31 @@ import { Schema } from "./schema";
  * @class
  */
 export class ManifestCoder {
-    private schema: Schema
+    private readonly elementDescriptor: ElementDescriptor;
 
-    constructor(elements: Map<number, LogicManifest.Element>, classDefs: Map<string, number>) {
-        this.schema = new Schema(elements, classDefs)
+    /**
+     * Creates an instance of ManifestCoder.
+     * 
+     * @param {LogicManifest.Manifest} elementDescriptor - The element descriptor.
+     * @constructor
+     */
+    constructor(elementDescriptor: ElementDescriptor);
+    /**
+     * Creates an instance of ManifestCoder.
+     * 
+     * @param {LogicManifest.Manifest} manifest - The logic manifest.
+     * @constructor
+     */
+    constructor(manifest: LogicManifest.Manifest);
+    /**
+     * Creates an instance of ManifestCoder.
+     */
+    constructor(descriptorOrManifest: LogicManifest.Manifest | ElementDescriptor) {
```

Comment:
can we accept `LogicManifest.Manifest` alone. I couldn't see any other plance where we are trying to pass element descriptor directly.